l10n_nl_xaf_auditfile_export/auditfile_export.py

```diff
@@ -0,0 +1,290 @@
+        for export in self.browse(cr, uid, ids):
+            if export.period_start.date_start > export.period_stop.date_stop:
+                return False
+        return True
+
+    # template (default)
+    def _default_template(self, cr, uid, ids):
+        self._logger.info('search for build-in xml_template')
+
+        _file = get_module_resource(
+            'l10n_nl_xaf_auditfile_export', 'data', 'auditfile_template.xml')
+        if not _file:
+            raise osv.except_osv(_('Data Error'), _('Invalid build-in template'))
+
+        return open(_file).read()
+
+    _defaults = {
+        'template_content': _default_template,
+        'name': 'auditfile %s' % datetime.datetime.now().date(),
+        'period_start': 1,
+        'period_stop': 12,
+    }
+
+    _constraints = [(_check_dates, _('Please, check periods'),
+                    ['period_start', 'period_stop'])]
+
+    # creates auditfile
+    def create_auditfile(self, cr, uid, ids, id):
+        self._logger.info('create auditfile')
+
+        _date_format = '%Y-%m-%d'
+        _output_file = '.xaf'
+
+        data = self.browse(cr, uid, ids[0], context=id)
+
+        # partners search
+        partners_obj = self.pool.get("res.partner")
+        partners_obj_ids = partners_obj.search(
+            cr, uid, ['|', ('supplier', '=', 'True'), ('customer', '=', 'True')]
+        )
+        partners = partners_obj.browse(cr, uid, partners_obj_ids)
+
+        if not partners:
+            raise osv.except_osv(_('Data Error'), _('There are no partners to export'))
+
+        def _validate_country_ids(partners):
+            names = []
+            if not partners[0].company_id.country_id:
+                names.append(partners[0].company_id.name)
+            for partner in partners:
+                if not partner.country_id:
+                    names.append(partner.name)
+                    break
+            if names:
+                raise osv.except_osv(
+                    _('Data Error'), _('Following companies have no country code: %s')
+                    % (', '.join(names))
+                )
+            return True
+        _validate_country_ids(partners)
+
+        # periods search
+        period_obj = self.pool.get("account.period")
+        period_ids = period_obj.search(
+            cr, uid, ['&', ('id', '>=', data.period_start.id), ('id', '<=', data.period_stop.id)]
+        )
+        periods = period_obj.browse(cr, uid, period_ids)
+
+        # journal search
+        journal_obj = self.pool.get("account.journal")
+        journal_ids = journal_obj.search(cr, uid, [])
+        journals = journal_obj.browse(cr, uid, journal_ids)
+
+        # accounts search
+        account_obj = self.pool.get("account.account")
+        account_ids = account_obj.search(cr, uid, [])
+        accounts = account_obj.browse(cr, uid, account_ids)
+        coa_ids = account_obj.search(cr, uid, ['&', ('user_type', '=', 1), ('parent_id', '=', False)])
```

Comment:
Replace hardcoded account_account_type id by xml id or something else that is more reliable.